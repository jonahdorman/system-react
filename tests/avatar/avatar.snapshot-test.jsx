/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { testDOMandHTML } from '../snapshot-helpers';

import SnapshotBase from '../../examples/avatar/base';
import SnapshotEntityIcon from '../../examples/avatar/entity-icon';
import SnapshotEntityInitials from '../../examples/avatar/entity-initials';
import SnapshotUserIcon from '../../examples/avatar/user-icon';
import SnapshotUserInitials from '../../examples/avatar/user-initials';

testDOMandHTML({
	name: 'Avatar Base',
	test,
	Component: SnapshotBase
});

testDOMandHTML({
	name: 'Avatar Entity Icon',
	test,
	Component: SnapshotEntityIcon
});

testDOMandHTML({
	name: 'Avatar Entity Initials',
	test,
	Component: SnapshotEntityInitials
});

testDOMandHTML({
	name: 'Avatar User Icon',
	test,
	Component: SnapshotUserIcon
});

testDOMandHTML({
	name: 'Avatar User Initials',
	test,
	Component: SnapshotUserInitials
});
