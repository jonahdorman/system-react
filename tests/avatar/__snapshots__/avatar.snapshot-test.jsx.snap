// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Base DOM Snapshot 1`] = `
<IconSettings
  iconPath="/assets/icons"
>
  <Avatar
    assistiveText="Avatar image"
    imgAlt="Person Name"
    imgSrc="https://lightningdesignsystem.com/assets/images/avatar2.jpg"
    size="medium"
    title="user avatar"
    variant="user"
  />
</IconSettings>
`;

exports[`Avatar Base HTML Snapshot 1`] = `"<div><span class=\\"slds-avatar slds-avatar_circle slds-avatar_medium\\"><img alt=\\"Person Name\\" src=\\"https://lightningdesignsystem.com/assets/images/avatar2.jpg\\" title=\\"user avatar\\"/></span></div>"`;

exports[`Avatar Entity Icon DOM Snapshot 1`] = `
<IconSettings
  iconPath="/assets/icons"
>
  <Avatar
    assistiveText="Account icon avatar"
    imgAlt=""
    size="medium"
    title="Avatar entity icon"
    variant="entity"
  />
</IconSettings>
`;

exports[`Avatar Entity Icon HTML Snapshot 1`] = `
"<div><span class=\\"slds-avatar slds-avatar_medium\\"><span class=\\"slds-icon_container slds-icon-standard-account\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon\\"><use xlink:href=\\"/assets/icons/standard-sprite/svg/symbols.svg#account\\"></use></svg><span class=\\"slds-assistive-text\\">Account icon avatar</span></span>
  </span>
</div>"
`;

exports[`Avatar Entity Initials DOM Snapshot 1`] = `
<IconSettings
  iconPath="/assets/icons"
>
  <Avatar
    imgAlt=""
    label="Acme Communications"
    size="medium"
    title="user avatar"
    variant="entity"
  />
</IconSettings>
`;

exports[`Avatar Entity Initials HTML Snapshot 1`] = `"<div><span class=\\"slds-avatar slds-avatar_medium\\"><abbr class=\\"slds-avatar__initials slds-icon-standard-account\\" title=\\"Acme Communications\\">AC</abbr></span></div>"`;

exports[`Avatar User Icon DOM Snapshot 1`] = `
<IconSettings
  iconPath="/assets/icons"
>
  <Avatar
    assistiveText="User Icon Avatar"
    imgAlt=""
    size="medium"
    title="User Icon Avatar"
    variant="user"
  />
</IconSettings>
`;

exports[`Avatar User Icon HTML Snapshot 1`] = `
"<div><span class=\\"slds-avatar slds-avatar_circle slds-avatar_medium\\"><span class=\\"slds-icon_container slds-icon-standard-user\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon\\"><use xlink:href=\\"/assets/icons/standard-sprite/svg/symbols.svg#user\\"></use></svg><span class=\\"slds-assistive-text\\">User Icon Avatar</span></span>
  </span>
</div>"
`;

exports[`Avatar User Initials DOM Snapshot 1`] = `
<IconSettings
  iconPath="/assets/icons"
>
  <Avatar
    imgAlt=""
    label="Annie Wilson"
    size="medium"
    title="user avatar"
    variant="user"
  />
</IconSettings>
`;

exports[`Avatar User Initials HTML Snapshot 1`] = `"<div><span class=\\"slds-avatar slds-avatar_circle slds-avatar_medium\\"><abbr class=\\"slds-avatar__initials slds-icon-standard-user\\" title=\\"Annie Wilson\\">AW</abbr></span></div>"`;
